pkg/cloudinit/controlplane_join_test.go

```diff
@@ -82,6 +83,86 @@ func TestNewJoinControlPlane(t *testing.T) {
 	), "Some /capi/scripts files are missing")
 }
 
+func TestNewJoinControlPlaneWithOptionalProxies(t *testing.T) {
+	g := NewWithT(t)
+
+	config, err := cloudinit.NewJoinControlPlane(cloudinit.JoinControlPlaneInput{
+		BaseUserData: cloudinit.BaseUserData{
+			KubernetesVersion: "v1.30.0",
+			BootCommands:      []string{"bootcmd"},
+			PreRunCommands:    []string{"prerun1", "prerun2"},
+			PostRunCommands:   []string{"postrun1", "postrun2"},
+			ExtraFiles: []cloudinit.File{{
+				Path:        "/tmp/file",
+				Content:     "test file",
+				Permissions: "0400",
+				Owner:       "root:root",
+			}},
+			SnapstoreProxyScheme: "http",
+			SnapstoreProxyDomain: "snapstore.io",
+			SnapstoreProxyID:     "abcd-1234-xyz",
+			HTTPProxy:            "http://proxy.internal",
+			HTTPSProxy:           "https://proxy.internal",
+			NoProxy:              "10.0.0.0/8,10.152.183.1,192.168.0.0/16",
+			ConfigFileContents:   "### config file ###",
+			MicroclusterAddress:  "10.0.0.11",
+		},
+		JoinToken: "test-token",
+	})
+
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// Verify the boot commands.
+	g.Expect(config.BootCommands).To(Equal([]string{"bootcmd"}))
+
+	// Verify the run commands.
+	g.Expect(config.RunCommands).To(Equal([]string{
```

Comment:
It's not necessary to re-test all these items. Only test the relevant ones since we have a test for all of them already. 

Instead, only test the items that would be affected by setting a containerd proxy . For example, test that the containerd proxy setup script is present